crates/mpz-ot-core/src/chou_orlandi/receiver.rs

```diff
@@ -86,125 +76,137 @@ impl Receiver {
-            compute_decryption_keys(sender_base_table, &private_keys, choices, *counter);
+        let (blinded_choices, new_keys) =
+            compute_decryption_keys(sender_base_table, &private_keys, &choices, *counter);
 
         *counter += blinded_choices.len();
-        cached_decryption_keys.extend(decryption_keys);
+        decryption_keys.extend(new_keys);
 
-        // If configured, log the choices
-        if self.config.receiver_commit() {
-            choice_log.extend(choices.iter_lsb0());
-        }
-
-        ReceiverPayload {
-            id: self.state.transfer_id,
-            blinded_choices,
-        }
+        ReceiverPayload { blinded_choices }
     }
 
-    /// Receives the encrypted payload from the Sender, returning the plaintext messages corresponding
-    /// to the Receiver's choices.
+    /// Receives the encrypted payload from the Sender.
     ///
     /// # Arguments
     ///
     /// * `payload` - The encrypted payload from the Sender
-    pub fn receive(&mut self, payload: SenderPayload) -> Result<Vec<Block>, ReceiverError> {
+    pub fn receive(&mut self, payload: SenderPayload) -> Result<()> {
         let state::Setup {
-            transfer_id: current_id,
+            transfer_id,
             decryption_keys,
             ..
         } = &mut self.state;
 
-        let SenderPayload { id, payload } = payload;
-
-        // Check that the transfer id matches
-        let expected_id = current_id.next();
-        if id != expected_id {
-            return Err(ReceiverError::IdMismatch(expected_id, id));
-        }
+        let SenderPayload { payload } = payload;
 
-        // Check that the number of ciphertexts does not exceed the number of pending keys
+        // Check that the number of ciphertexts does not exceed the number of pending
+        // keys
         if payload.len() > decryption_keys.len() {
             return Err(ReceiverError::CountMismatch(
                 decryption_keys.len(),
                 payload.len(),
             ));
         }
 
-        // Drain the decryption keys and decrypt the ciphertexts
-        Ok(decryption_keys
-            .drain(..payload.len())
-            .zip(payload)
-            .map(
-                |((c, key), [ct0, ct1])| {
-                    if c {
-                        key ^ ct1
-                    } else {
-                        key ^ ct0
-                    }
-                },
-            )
-            .collect::<Vec<Block>>())
+        let mut msgs =
+            decryption_keys
+                .drain(..payload.len())
+                .zip(payload)
+                .map(
+                    |((c, key), [ct0, ct1])| {
+                        if c {
+                            key ^ ct1
+                        } else {
+                            key ^ ct0
+                        }
+                    },
+                );
+
+        while let Some(Queued { count, sender }) = self.queue.pop_front() {
```

Comment:
@sinui0 
Sorry, I'll try to more accurately describe the issue because your response didn't seem to address it.

Suppose, the receiver queues 60 OTs.
Suppose in `receive` the length of `SenderPayload` is 30.

Why is it not an error that the receiver will get only 30 OTs out of the 60 that they intended to receive?